typegen/src/utils.rs

```diff
@@ -140,52 +155,68 @@ fn types_equal_inner(
         }
     }
 
-    let a_ty = types.resolve(a).expect("type a should exist in registry");
-    let b_ty = types.resolve(b).expect("type b should exist in registry");
-
     // Paths differ; types won't be equal then!
     if a_ty.path.segments != b_ty.path.segments {
         return false;
     }
 
-    // Names of type params don't line up, so different then!
-    if a_ty
-        .type_params
-        .iter()
-        .zip(&b_ty.type_params)
-        .any(|(a, b)| a.name != b.name)
-    {
-        return false;
-    }
+    let mut compare_fields = |a_params: &GenericsList,
+                              b_params: &GenericsList,
+                              a: &Field<PortableForm>,
+                              b: &Field<PortableForm>|
+     -> bool {
```

Comment:
Perhaps we should `#[rustfmt::skip]` on this like the one below, because this formatting is hideous :)

Also, for consistency with below, maybe stick to same param order ie a,a_params,b,b_params (or make the other match this)?

Reply:
reformatted and reordered params as suggested